crates/evm/src/execute.rs

```diff
@@ -252,7 +268,13 @@ pub trait BlockExecutionStrategyFactory: ConfigureEvmFor<Self::Primitives> + 'st
     ///
     /// This is similar to [`alloy_evm::EvmEnv`], but only contains context unrelated to EVM and
     /// required for execution of an entire block.
-    type ExecutionCtx<'a>;
+    type ExecutionCtx<'a>: Clone;
+
+    /// A type that knows how to build a block.
+    type BlockFactory: BlockFactory<Self>;
+
+    /// Provides reference to configured [`BlockFactory`].
+    fn block_factory(&self) -> &Self::BlockFactory;
```

Comment:
this is nice.

but now we have a quite a few things here, and we should add some docs here that explain what the purpose of this is and how this differs from the strategy itself.